test/assertions.js

```diff
@@ -100,6 +100,24 @@ exports.mixinMochaAssertions = function(expect) {
         code: expect.it('to be greater than', 0)
       });
     })
+    .addAssertion(
+      '<RawRunResult|RawResult> [not] to have failed (with|having) output <any>',
+      function(expect, result, output) {
+        expect(result, '[not] to satisfy', {
+          code: expect.it('to be greater than', 0),
+          output: output
+        });
```

Comment:
OK. See your rule, but don't see how it would be satisfied.
```js
// simulate output from [_spawnMochaWithListeners](https://github.com/mochajs/mocha/blob/c77d396e7607289a40c2cf5713403a2ed72c9bf9/test/integration/helpers.js#L261)
var res = {
  output: ['{ "foo": 123, "bar": true, "baz": "abc" }'],
  code: 1,
  args: '--whatever'
};
// convert run result from [toJSONRunResult](https://github.com/mochajs/mocha/blob/c77d396e7607289a40c2cf5713403a2ed72c9bf9/test/integration/helpers.js#L186)
var result = toJSONRunResult(res);
// => { foo: 123, bar: true, baz: 'abc', code: 1 }
// fn(null, result);

It still has `code`, but wasn't `output` lost in translation?
```

Reply:
hmm, I still don't really follow.  this does not apply to `JSONRunResult`; it can't be used with `runMochaJSON`.  generally when we use `runMochaJSON` we're more interested in specific information, e.g., did test _x_ pass and how many pending tests were there.

Reply:
I see now. Assumed this was related to your change in "test/integration/helpers.js", but it's unrelated to the other unrelated changes of reverting the original PR...